UnitTests/GitUI.Tests/UserControls/RevisionGrid/RevisionFileNameTests.cs

```diff
@@ -0,0 +1,78 @@
+using NSubstitute;
+
+namespace GitUITests.UserControls.RevisionGrid;
+
+[TestFixture]
+[Apartment(ApartmentState.STA)]
+public class RevisionFileNameTests
+{
+    private RevisionGridControl _revisionGridControl;
+    private MockExecutable _executable;
+
+    [SetUp]
+    public void SetUp()
+    {
+        _revisionGridControl = new RevisionGridControl();
+        _executable = new MockExecutable();
+
+        ServiceContainer serviceContainer = new();
+        serviceContainer.AddService(Substitute.For<GitUI.Hotkey.IHotkeySettingsManager>());
+        serviceContainer.AddService(Substitute.For<ResourceManager.IHotkeySettingsLoader>());
+
+        IGitUICommandsSource uiCommandsSource = Substitute.For<IGitUICommandsSource>();
+        GitModule? gitModule = GetGitModuleWithExecutable(executable: _executable);
+
+        GitUICommands uiCommands = new(serviceContainer, gitModule);
+        uiCommandsSource.UICommands.Returns(_ => uiCommands);
+        _revisionGridControl.UICommandsSource = uiCommandsSource;
+        _revisionGridControl.FilePathByObjectId = new();
+    }
+
+    [TestCase("17b2a8777e43dff588284c9661206c97ccb6cf8e", "a.txt")]
+    [TestCase("83ec7c0a516f00b67aa9915d8e673a26510c2eed", "b.txt")]
+    public void GetRevisionFileName_should_query_git_and_return_file_name(string objectId, string expectedFileName)
+    {
+        const string path = "a.txt";
+        string actualFileName;
+        string line = $"-c log.showSignature=false log --format=\"????%H\" --name-only --follow --diff-merges=separate  --find-renames --find-copies {objectId}^1..{objectId} -- a.txt";
+
+        bool originalFollowRenamesInFileHistoryExactOnly = AppSettings.FollowRenamesInFileHistoryExactOnly;
+        AppSettings.FollowRenamesInFileHistoryExactOnly = false;
+
+        try
+        {
+            using (_executable.StageOutput(line, $"????{objectId}\n{expectedFileName}"))
+            {
+                actualFileName = _revisionGridControl.GetRevisionFileName(path, ObjectId.Parse(objectId));
+            }
+        }
+        finally
+        {
+            AppSettings.FollowRenamesInFileHistoryExactOnly = originalFollowRenamesInFileHistoryExactOnly;
+        }
+
+        Assert.AreEqual(expectedFileName, actualFileName);
+    }
+
+    /// <summary>
+    /// Create a GitModule with mockable GitExecutable
+    /// </summary>
+    /// <param name="executable">The mock executable</param>
+    /// <returns>The GitModule</returns>
+    private static GitModule GetGitModuleWithExecutable(IExecutable executable)
+    {
+        GitModule? module = new(null);
+
+        typeof(GitModule).GetField("_gitExecutable", BindingFlags.Instance | BindingFlags.NonPublic)?.SetValue(module, executable);
+        GitCommandRunner cmdRunner = new(executable, () => GitModule.SystemEncoding);
+        typeof(GitModule).GetField("_gitCommandRunner", BindingFlags.Instance | BindingFlags.NonPublic)?.SetValue(module, cmdRunner);
+
+        return module;
```

Comment:
Is there any reason not to use `MockExecutable`?
https://github.com/gitextensions/gitextensions/blob/eeaf3c13e644a15178686af70bf362547943daa6/UnitTests/GitCommands.Tests/Git/Gpg/GitGpgControllerTests.cs#L16-L25

Reply:
I tried but I cannot get it through the whole hierarchy of mock objects. Specifically, I need a `GitUICommands` instance (**not** `IGitUiCommands`) for the `IGitUICommandsSource.UICommands` property. But the constructor of `GitUICommands` takes a `GitModule` (**not** `IGitModule`) as a parameter.

Reply:
`IGitModule` has been completed. Perhaps, the needed things become available after resolving the MC.

Reply:
Please rebase on top of the latest master. Since https://github.com/gitextensions/gitextensions/pull/11269 all public API use `IGitModule`.